ComponentFramework3.0/ComponentFramework/FileManager.cpp

```diff
@@ -0,0 +1,175 @@
+#include "FileManager.h"
+#include <vector>
+#include "json.hpp"
+#include <fstream>
+#include <iostream>
+#include "Scene0.h"
+#include "GameObject.h"
+
+/*
+
+the file manager  should have empty vector structer  what need to do is dependeding on the
+situtation it will read and write or it will take a refernce from the into a file
+
+
+
+
+*/
+using json = nlohmann::json;
+
+using namespace GAME;
+using namespace MATH;
+
+FileManager::FileManager()
+{
+
+}
+
+
+FileManager::~FileManager()
+{
+
+}
+
+void FileManager::OnWrite(std::vector<GameObject*> ScenceModelList)
+{
+	
+
+	std::ofstream info;
+	//creates a file called Savedata.txt
+	info.open("Savedata.txt");
```

Comment:
avoid using raw strings in the body of your code, make a constant and give it a relevant name instead.
```
const string exportFileName = "Savedata.txt"
```

Reply:
Then in your code you can use it like
```
info.open(kSaveFile)
```
and you wont have to worry about spelling it correctly ever again, just use the constant from then on.